psycop/common/feature_generation/sequences/timeseries_windower_python/patient.py

```diff
@@ -0,0 +1,63 @@
+import datetime as dt
+from dataclasses import dataclass
+from typing import Sequence
+
+from psycop.common.feature_generation.sequences.timeseries_windower_python.events.static_event import (
+    StaticEvent,
+)
+from psycop.common.feature_generation.sequences.timeseries_windower_python.events.temporal_event import (
+    TemporalEvent,
+)
+from psycop.common.feature_generation.sequences.timeseries_windower_python.prediction_sequence import (
+    PredictionSequence,
+)
+
+
+@dataclass(frozen=True)
+class Patient:
+    """All task-agnostic data for a patient."""
+
+    patient_id: str | int
+    temporal_events: Sequence[TemporalEvent]
+    # TODO: Should we check that temporal events are sorted by timestamp on init?
+
+    static_events: Sequence[StaticEvent] | None
+
+    @staticmethod
+    def _filter_events(
+        events: Sequence[TemporalEvent], start: dt.datetime, end: dt.datetime
+    ) -> Sequence[TemporalEvent]:
+        # This could be much faster if we assume that the events are already sorted.
+        # Then we could implement binary search, which is O(log n) instead of O(n).
+        # However, this might be plenty fast. We can always optimize later.
+        return [event for event in events if start <= event.timestamp < end]
```

Comment:
If we assume sorted when should just sort on object initialization?

Should the filter method be on patient (I could imagine also using it on other lists of events). Generally I would add a filter method to the patient which is arbitrary.

```suggestion
    def filter_events(
        self, filter: Callable,
    ):
```